frontend/libs/studio-components-legacy/src/components/StudioCodelistEditor/StudioCodeListEditorReducer.test.tsx

```diff
@@ -1,50 +1,74 @@
-import { ObjectUtils } from '@studio/pure-functions';
 import { codeListWithTextResources } from './test-data/codeListWithTextResources';
 import { textResources } from './test-data/textResources';
-import { reducer, ReducerActionType } from './StudioCodeListEditorReducer';
 import type { ReducerAction, ReducerState } from './StudioCodeListEditorReducer';
+import { reducer, ReducerActionType } from './StudioCodeListEditorReducer';
 import { CodeListItemTextProperty } from './types/CodeListItemTextProperty';
 import type { TextResource } from '../../types/TextResource';
+import type { CodeList } from './types/CodeList';
 
 // Test data:
 const defaultState = {
   codeList: codeListWithTextResources,
   textResources: textResources,
 };
 
-const dispatch = (action: ReducerAction, state?: Partial<ReducerState>) => {
-  return reducer(state ? { ...defaultState, ...state } : { ...defaultState }, action);
+const dispatch = (action: ReducerAction, state?: Partial<ReducerState>): ReducerState => {
+  return reducer({ ...defaultState, ...state }, action);
 };
 
 describe('StudioCodeListEditorReducer', () => {
-  afterEach(jest.clearAllMocks);
-
   describe('SetCodeList', () => {
-    it('should update state correctly', () => {
-      const testCodeList = [{ value: 'test1', label: 'test2' }];
-      const expectedState = { codeList: testCodeList, textResources: textResources };
+    it('should update codeList property in state correctly', () => {
+      const testCodeList: CodeList = [{ value: 'test1', label: 'test2' }];
 
       const state: ReducerState = dispatch({
         type: ReducerActionType.SetCodeList,
         codeList: testCodeList,
       });
 
-      expect(state).toEqual(expectedState);
+      expect(state.codeList).toEqual(testCodeList);
+    });
+  });
+
+  describe('setTextResources', () => {
+    it('should update textResources property in state correctly', () => {
+      const testTextResources: TextResource[] = [{ id: 'test1', value: 'test2' }];
+
+      const state: ReducerState = dispatch({
+        type: ReducerActionType.SetTextResources,
+        textResources: testTextResources,
+      });
+
+      expect(state.textResources).toEqual(testTextResources);
     });
   });
 
   describe('AddTextResource', () => {
-    it('should update state correctly when adding text resource for label', () => {
+    it('should update textResources property in state correctly when adding text resource for label', () => {
```

Comment:
Småpirk: Lag gjerne beskrivelser som sier mer tydelig hva testen sjekker. Fraser som "updates correctly" kan gi mening ved testing av mer sammensatt data (som ved forrige kodegjennomgang), men denne testen tester jo nå én enkel ting. Eksempel:
```suggestion
    it('Adds the new text resource to the list of text resources', () => {
```
(Vi har ingen konvensjon på det, men jeg pleier å skrive i presens for å gjøre beskrivelsene mer konsise. Det blir litt overflødig med `should` overalt. Men jeg skal ikke legge noen føringer om det siden `should` er såpass etablert.)

Å spesifisere at det er "for label" tenker jeg også er unødvendig. `CodeListItemTextProperty.Label` er bare en del av testdataen, men vi vil jo verifisere at det fungerer uavhengig av hvilken egenskap det gjelder. Det kan være gode grunner til å teste spesifikt for en gitt egenskap, men da bør vi også ha tester for de andre egenskapene.